packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneAndTalkButton/DoneAndTalkButtonContainer.js

```diff
@@ -25,29 +29,51 @@ function storeMapper(stores) {
 @observer
 class DoneAndTalkButtonContainer extends React.Component {
   render () {
-    const { 
+    const {
       completed,
       demoMode,
       disabled,
       goldStandardMode,
-      onClick, 
-      project, 
-      shouldWeShowDoneAndTalkButton, 
-      subject 
+      onClick,
+      project,
+      setOnHide,
+      shouldWeShowDoneAndTalkButton,
+      subject
     } = this.props
     const projectSlug = project && project.slug
     const subjectId = subject && subject.id
 
     if (shouldWeShowDoneAndTalkButton && projectSlug && subjectId) {
       const talkURL = `/projects/${projectSlug}/talk/subjects/${subjectId}`
 
+      function openTalkLinkAndClick (event) {
+        const isCmdClick = event.metaKey
+
+        onClick(event)
+          .then(() => {
+            if (window && talkURL) {
+              const url = `${window.location.origin}${talkURL}`
+              if (isCmdClick) {
+                event.preventDefault()
+                const newTab = window.open()
+                newTab.opener = null
+                newTab.location = url
+                newTab.target = '_blank'
+                newTab.focus()
+              } else {
+                setOnHide(() => window.location.assign(url))
```

Comment:
This call could be tested.

Reply:
Mocking `window.location` turned out to be more complicated than I thought it would be. I'm not sure if there's an easier way to test this with JSDOM.

Reply:
JSDOM doesn't really support it: https://github.com/jsdom/jsdom/issues/2112

Some seem to have gotten hacky workarounds working, but that seems it'd just be making a brittle test to me.

Reply:
I think that might have changed in the same update that added local storage, because JSDOM throws a completely different error now. Navigation is supported, but you can't alter those methods by wrapping them.

Reply:
You could pass in a function calling `window.location` as a default prop and pass in a spy in the tests?

Reply:
At the moment, I've replaced `window.location.assign` with a spy.
https://github.com/zooniverse/front-end-monorepo/pull/599/files#diff-b5b3fa22ac8afaf7491453ab3504e35dR66